### ISSUE TITLE
Bug in random-max-percent calculation

### ISSUE THREAD

**Issue by gzlatkov** [asobti/kube-monkey#81]
I believe there is a bug in:
https://github.com/asobti/kube-monkey/blob/e1052dfc639814fdc9919519eb38d82b8e7155e7/chaos/chaos.go#L125

According to the `README`, `killValue` should be between 0-100. Therefore, the value passed to `DeleteRandomPods` will be between 0 and 10,000 (`100*100/(0+1)`).

Example: 
https://play.golang.org/p/c0z6RijiR1D

Think one way to calculate the correct value is:
```
nPods := 3
maxPods := math.Round(float64(nPods*killValue) / 100)
toKill := r.Intn(int(maxPods+1))
```

Happy to submit a PR with a fix and some tests.

**Comment by asobti** [comment]
@gzlatkov Thanks for catching this. This does seem like a bug. A PR would be really appreciated.

A couple of thoughts:
1. Instead of `math.Round()`, lets do `math.Floor()` instead so we never over-kill pods. So, with `n=3` and `killValue=50`, we should never kill 2 pods.
2. I'm not sure what the best way to get the number of pods in `chaos.go` is. That info isn't already available there, so it will likely need to be added to the `Victim` interface so each victim can report the number of pods.

cc @Spellchaser

**Comment by gzlatkov** [comment]
Hey @asobti,

1. That sounds reasonable!
2. I think we do have that information as it is extracted from the `clientset` which we are already have in `func (c *Chaos) terminate(clientset kube.Interface) error`. Will try to test that tomorrow.

**Comment by asobti** [comment]
Hey @gzlatkov . Just checking in to see if you had a chance to work on this bug.

**Comment by worldtiki** [comment]
Hi @asobti. I was looking into adding a fixed percentage mode when I came across this same issue.
I raised a pr with the my proposal for a new kill mode which includes a fix for this bug: https://github.com/asobti/kube-monkey/pull/94

**Comment by gzlatkov** [comment]
@asobti Sorry, got caught up in work and was on holiday. My stuff still need the unittest adjustments, so I'm happy to go over @worldtiki's PR, which seems to be expanding on the fix with a new termination mode.

**Comment by asobti** [comment]
Fixed by #94
